Source/Android/jni/AndroidCommon/IDCache.cpp

```diff
@@ -29,9 +29,27 @@ static jmethodID s_do_rumble;
 
 namespace IDCache
 {
-JavaVM* GetJavaVM()
-{
-  return s_java_vm;
+JNIEnv* GetEnvForThread()
+{
+  thread_local static struct owned_env

```

Comment:
Sorry, maybe I wasn't clear with my earlier comment. The struct name should still be `OwnedEnv`. The underscore style of naming is only used for variable names.